### ISSUE TITLE
QCGPUSimulatorError: 'Measurements are only supported at the end'

### ISSUE THREAD

**Issue by cogig** [Qiskit/qiskit-qcgpu-provider#6]
![image](https://user-images.githubusercontent.com/45864384/51792214-11214380-21e9-11e9-8047-ebf6c9d7803d.png)

**Comment by adamisntdead** [comment]
Hello!

Can you tell me what circuit you were trying to run?

**Comment by cogig** [comment]
The circuit image size is oversize and worrisome.I find that after i remove all "reset" operation.It works well.But in my circuit ,"reset" operation is very import .How can I fix the error?

**Comment by cogig** [comment]
https://qiskit.org/documentation/summary_of_quantum_operations.html#reset
![image](https://user-images.githubusercontent.com/45864384/51792490-82aec100-21ec-11e9-91e5-b2cabcf77852.png)

**Comment by cogig** [comment]
Is there a temporary solution to fix the problem?

**Comment by adamisntdead** [comment]
Is it possible to do the reset operations once you have got the results?

**Comment by pittcat** [comment]
Because i indend to realize one quantum circuit multiplexing to improve efficiency.And in the end, we can get the data that we need.

**Comment by adamisntdead** [comment]
Okay, I will try add the resets, give me a few minutes

**Comment by pittcat** [comment]
@adamisntdead In our project,the number of quantum circuit is our primary consideration.So the reutilization quantum circuit  is very import.

**Comment by adamisntdead** [comment]
Sorry, can you rephrase?

**Comment by cogig** [comment]
Oh,many thanks.

**Comment by cogig** [comment]
Ok,you can establish a new branch.

**Comment by cogig** [comment]
After I removed old packages,I cloned the both qiskit-qcgpu-provider and qcgpu to home and installed the two packages successfully by "python3 setup.py install --user".The same error take place when i add "reset" operation to the circuit.There may be something wrong by my incorrect operation.But I can't dope out a solution to the problem.

**Comment by cogig** [comment]
```
# In[3]:


import numpy as np
from qiskit.tools.visualization import plot_histogram
from qiskit import QuantumCircuit, ClassicalRegister, QuantumRegister
from qiskit import execute, BasicAer
from qiskit_qcgpu_provider import QCGPUProvider


# In[4]:


col=2
state=6
reg_counts = 10

qr = QuantumRegister(reg_counts, 'qr')
cr = ClassicalRegister(reg_counts, 'cr')
circuit = QuantumCircuit(qr)
measurement_circuit=QuantumCircuit(cr)
for i in range(col, state):
    circuit.h(qr[i])
circuit.ccx(qr[8],qr[9],qr[0])
circuit.cx(qr[8],qr[1])


# In[5]:


circuit.reset(qr[6])


# In[6]:



Provider = QCGPUProvider()

circuit.add_register(cr)
circuit.measure(qr,cr)



print("QCGPU backends: ", Provider.backends())
backend_sim = Provider.get_backend('qasm_simulator')


job_sim = execute(circuit, backend_sim)
result_sim = job_sim.result()




counts=result_sim.get_counts(circuit)
counts


all_lst=list(counts.keys())
all_lst

```
If deleting the line that contains "reset" operation ,the error will fade away and you can check the code

**Comment by adamisntdead** [comment]
Okay, I will have a look at this tonight
